mlflow/sklearn/__init__.py

```diff
@@ -996,6 +1397,13 @@ def eval_and_log_metrics(model, X, y_true, *, prefix, sample_weight=None):
       - prefix is empty
       - model is not an sklearn estimator or does not support the 'predict' method
     """
+    status = _get_autolog_training_status()
+    with status.disable_log_post_training_metrics():
```

Comment:
Can we add an inline comment explaining why we're disabling the logging of post training metrics?